pftp/client_handler.go

```diff
@@ -64,17 +72,33 @@ func newClientHandler(connection net.Conn, c *config, m middleware, id int, curr
 		context:           newContext(c),
 		currentConnection: currentConnection,
 		mutex:             &sync.Mutex{},
-		connMutex:         cMutex,
+		readlockMutex:     &sync.Mutex{},
 		log:               &logger{fromip: connection.RemoteAddr().String(), id: id},
 		srcIP:             connection.RemoteAddr().String(),
 		tlsProtocol:       0,
 		isLoggedin:        false,
-		chkEstablished:    clientEstablished,
+		readLock:          false,
+		gotResponse:       make(chan struct{}),
+		isDone:            false,
+	}
+
+	// is masquerade IP not setted, set local IP of client connection
+	if len(p.config.MasqueradeIP) == 0 {
+		p.config.MasqueradeIP = strings.Split(connection.LocalAddr().String(), ":")[0]
 	}
 
 	return p
 }
 
+func (c *clientHandler) Close() error {
+	if c.conn != nil {
+		err := c.conn.Close()
```

Comment:
```golang
return c.conn.Close()
```

Reply:
Oh, it's very strange to see now.

Thanks!